jobs/pacman-data-shipper/src/main/java/com/tmobile/cso/pacman/datashipper/entity/AssetGroupManager.java

```diff
@@ -0,0 +1,357 @@
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.common.collect.Lists;
+import com.google.common.collect.Maps;
+import com.tmobile.cso.pacman.datashipper.dao.RDSDBManager;
+import com.tmobile.cso.pacman.datashipper.es.ESManager;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.text.SimpleDateFormat;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.UUID;
+import java.util.stream.Collectors;
+
+public class AssetGroupManager {
+    private static final Logger LOGGER = LoggerFactory.getLogger(AssetGroupManager.class);
+    private static final String DATE_FORMAT = "MM/dd/yyyy HH:mm";
+    private static final String EMPTY_STRING = "";
+    private static final String ASSET_GROUP_FOR_ALL_RESOURCES = "all-resources";
+    private static final String FETCH_IMPACTED_ALIAS_QUERY_TEMPLATE = "SELECT DISTINCT agd.groupId, agd.groupName, " +
+            "agd.groupType, agd.aliasQuery FROM cf_AssetGroupDetails AS agd " +
+            //Checks if data source is selected as cloudType
+            "WHERE (agd.groupId NOT IN ( " +
+            "    SELECT DISTINCT agcd.groupId " +
+            "    FROM cf_AssetGroupCriteriaDetails AS agcd " +
+            "    WHERE agcd.attributeName = 'CloudType') AND agd.groupType <> 'user' AND agd.groupType <> 'System' " +
+            "AND agd.groupName <> 'all-cloud' and aliasQuery like '%s') " +
+            //for all user asset groups
+            "OR (agd.groupType = 'user') " +
+            //for current data source
+            "OR (agd.groupName = '%s' AND agd.groupType = 'System') " +
+            //for all-resources
+            "OR (agd.groupName = '" + ASSET_GROUP_FOR_ALL_RESOURCES + "') " +
+            "OR (agd.groupType <> 'user' " +
+            "    AND agd.groupType <> 'System' AND agd.groupName <> 'all-cloud' and aliasQuery like '%s')";
+    private static final String UPDATE_ES_ALIAS_TEMPLATE = "{\"actions\": [{\"add\": {%s \"index\": \"%s\"," +
+            "\"alias\": \"%s\"}}]}";
+    private static final String FILTER_TEMPLATE = "\"filter\": %s,";
+    private static final String ES_ATTRIBUTE_TAG = "tags.";
+    private static final String ES_ATTRIBUTE_KEYWORD = ".keyword";
+    private static final String DISTINCT_DATA_SOURCES_QUERY_TEMPLATE = "SELECT DISTINCT dataSourceName FROM cf_Target";
+    private static final String CHECK_PLUGIN_ENABLED_QUERY_TEMPLATE =
+            "SELECT `value` FROM pac_config_properties WHERE cfkey = '%s.enabled'";
+    private static final String CRITERIA_DETAILS_QUERY_TEMPLATE =
+            "SELECT attributeName, attributeValue FROM cf_AssetGroupCriteriaDetails WHERE groupId = '%s'";
+    private static final Map<String, List<String>> dbCache = new HashMap<>();
```

Comment:
Please add test results how faster this Manager class became after adding caching.

Reply:
added